src/app/shared-services/trajectory.service.ts

```diff
@@ -30,6 +30,8 @@ export class TrajectoryService {
     private locationService: LocationService
   ) {}
 
+  private trajectoryCache: Trajectory
```

Comment:
This breaks, as soon as there are multiple calls of `getOne()` before all callers have called `.unsubscribe()` on the result.
Make this a Map of Trajectories, where the keys are the trajectory ID?

Actually, thinking about it, we don't need that separate cache at all, as we can use the closure of `new Observable(...` below for that?